pkg/maps/lbmap/lbmap.go

```diff
@@ -313,27 +325,55 @@ func UpdateWrrSeq(fe ServiceKey, weights []uint16) error {
 	if err != nil {
 		return fmt.Errorf("unable to generate weighted round robin seq for %s with value %+v: %s", fe.String(), weights, err)
 	}
-	return UpdateServiceWeights(fe, svcRRSeq)
+	return updateServiceWeights(fe, svcRRSeq)
 }
 
-// AddSVC2BPFMap adds the given bpf service to the bpf maps.
-func AddSVC2BPFMap(fe ServiceKey, besValues []ServiceValue, addRevNAT bool, revNATID int) error {
-	var err error
-	var weights []uint16
-	// Put all the backend services first
-	nSvcs := 1
-	nNonZeroWeights := 0
+func updateMasterService(fe ServiceKey, nbackends int, nonZeroWeights uint16) error {
+	fe.SetBackend(0)
+	zeroValue := fe.NewValue().(ServiceValue)
+	zeroValue.SetCount(nbackends)
+	zeroValue.SetWeight(nonZeroWeights)
+
+	return updateService(fe, zeroValue)
+}
+
+// UpdateService adds or updates the given service in the bpf maps
+func UpdateService(fe ServiceKey, besValues []ServiceValue, addRevNAT bool, revNATID int) error {
+	var (
+		weights         []uint16
+		nNonZeroWeights uint16
+		existingCount   int
+	)
 
 	for _, be := range besValues {
-		fe.SetBackend(nSvcs)
 		weights = append(weights, be.GetWeight())
 		if be.GetWeight() != 0 {
 			nNonZeroWeights++
 		}
-		if err = UpdateService(fe, be); err != nil {
+	}
+
+	mutex.Lock()
+	defer mutex.Unlock()
+
+	svcValue, err := lookupService(fe)
```

Comment:
Do we still want to continue if err is not nil? I guess so, if the service does not exist. Can you add a comment here if that's the case?

Reply:
Added comment